### PR TITLE
Added e2e test for ipfs toggle

### PR DESCRIPTION
**Pull request by NidhiKJha**
This PR is to add e2e test for IPFS toggle and a follow up PR for this [20172](https://github.com/MetaMask/metamask-extension/pull/20172)

* Fixes [#1129](https://github.com/MetaMask/MetaMask-planning/issues/1129)

## Screencast

https://github.com/MetaMask/metamask-extension/assets/39872794/32bae19d-cd59-4146-8835-d449638dde5a

## Manual Testing Steps

- Check the e2e file `test/e2e/tests/ipfs-toggle.spec.js`

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [x] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

File: test/e2e/tests/ipfs-toggle.spec.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  const smartContract = SMART_CONTRACTS.ERC1155;
+  const ganacheOptions = {
+    accounts: [
+      {
+        secretKey:
+          '0x7C9529A67102755B7E6102D6D950AC5D5863C98713805CEC576B945B15B71EAC',
+        balance: convertToHexValue(25000000000000000000),
+      },
+    ],
+  };
```
Comment (DDDDDanica) [inline test/e2e/tests/ipfs-toggle.spec.js:16]: you can use `defaultGanachOptions`

File: test/e2e/tests/ipfs-toggle.spec.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        dapp: true,
+        fixtures: new FixtureBuilder().withNftControllerERC1155().build(),
+        ganacheOptions,
+        smartContract,
+        title: this.test.title,
+      },
+      async ({ driver }) => {
+        await driver.navigate();
+        await driver.fill('#password', 'correct horse battery staple');
+        await driver.press('#password', driver.Key.ENTER);
```
Comment (DDDDDanica) [inline test/e2e/tests/ipfs-toggle.spec.js:29]:
```suggestion
        await unlockWallet(driver):
```

File: test/e2e/tests/ipfs-toggle.spec.js (hunk 1 above)
Comment (NidhiKJha) [reply test/e2e/tests/ipfs-toggle.spec.js]: Updated here https://github.com/MetaMask/metamask-extension/pull/20360/commits/ea179ad09018fc212e9eb0a27cab98992bda3895

File: test/e2e/tests/ipfs-toggle.spec.js (hunk 2 above)
Comment (NidhiKJha) [reply test/e2e/tests/ipfs-toggle.spec.js]: Updated here https://github.com/MetaMask/metamask-extension/pull/20360/commits/ea179ad09018fc212e9eb0a27cab98992bda3895

**Review by pedronfigueiredo — APPROVED** [review]

**Comment by pedronfigueiredo** [comment]
LGTM!

**Review by DDDDDanica — APPROVED** [review]

**Comment by DDDDDanica** [comment]
LGTM!

**Pull request merged**